roles/ovirt-engine-setup/tasks/main.yml

```diff
@@ -0,0 +1,69 @@
+---
+# health page
+- name: check if ovirt-engine running (health page)
+  uri:
+    url: "http://{{ ansible_fqdn }}/ovirt-engine/services/health"
+    status_code: 200
+  register: ovirt_engine_status
+  retries: 2
+  delay: 5
+  until: ovirt_engine_status|success
+  ignore_errors: True
+
+# copy default answer file
+- name: copy default answerfile
+  template:
+    src: answerfile_{{ ovirt_engine_version }}_basic.txt.j2
+    dest: /tmp/answerfile.txt
+    mode: 0644
+    owner: root
+    group: root
+  when: ovirt_engine_answer_file_path is undefined
+
+# copy custom answer file
+- name: copy custom answer file
+  template:
+    src: "{{ ovirt_engine_answer_file_path }}"
+    dest: /tmp/answerfile.txt
+    mode: 0644
+    owner: root
+    group: root
+  when: ovirt_engine_answer_file_path is defined
+
+- name: update setup packages
+  yum:
+    name: 'ovirt*setup*'
+    state: latest
+  when: ovirt_engine_update
+  tags:
+    - skip_ansible_lint
+
+- name: run engine-setup with answerfile
+  shell: 'engine-setup --config-append=/tmp/answerfile.txt'
+  when: ovirt_engine_status|failed or ovirt_engine_update
+  tags:
+    - skip_ansible_lint
+
+- name: check state of engine
+  service:
+    name: ovirt-engine
+    state: started
+
+- name: restart of ovirt-engine service
+  service:
+    name: ovirt-engine
+    state: restarted
+
+- name: check health status of page
+  uri:
+    url: "http://{{ ansible_fqdn }}/ovirt-engine/services/health"
+    status_code: 200
+  register: health_page
+  retries: 12
+  delay: 10
+  until: health_page|success
+
+- name: clean tmp files
```

Comment:
I would make sure this is always removed to not store any password in plain in case of failure.

Reply:
i will change it to block and add the always section